### PR TITLE
Character list

### PR DESCRIPTION
**Pull request by AngieDutra**
This PR closes issue #926 
Refe-1

Check:
- you can go from MainScreen to this screen and back to Main
- the list scrolls as expected
- the information is right
- the pressed state reduces the element's scale

The only files with code to check are:
- client/Assets/Scripts/CoMCharacter.cs
- client/Assets/Scripts/UI/ButtonAnimations.cs
- client/Assets/Scripts/UI/CharactersListManager.cs

Note: this PR changes Lobbies.unity file due to UI problems solved when updating the back button arrow

### CODE REVIEW

File: client/ProjectSettings/EditorBuildSettings.asset [hunk 1]
```diff
@@ -167,4 +167,7 @@ EditorBuildSettings:
   - enabled: 1
     path: Assets/Scenes/TitleScreen.unity
     guid: 4900ba75a16aa4edbb73129a2a725d34
+  - enabled: 0
```
Comment (Nico-Sanchez) [inline client/ProjectSettings/EditorBuildSettings.asset:170]: I had an error with the character list scene and had to enable it from Unity.

I believe if you enable it here that error will be fixed. Can we do that or this is zero on purpose?

File: client/Assets/Scripts/UI/CharactersListManager.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    }
+
+    void GenerateList()
+    {
+        players.ForEach(player =>
+        {
+            GameObject item = (GameObject)Instantiate(listItem, this.transform);
+            item.GetComponentInChildren<Image>().sprite = player.characterSprite;
+            item.GetComponentInChildren<TextMeshProUGUI>().text = player.name;
+            //item.GetComponentInChildren<TextMeshPro>().text = player.name;
```
Comment (Nico-Sanchez) [inline client/Assets/Scripts/UI/CharactersListManager.cs:28]: Can we remove this comment? Do we want to keep this?

**Review by Nico-Sanchez — CHANGES_REQUESTED** [review]
The character list scene looks good. Nice job!

Left some comments regarding the code and an error I had when testing it.

File: client/Assets/Scripts/UI/CharactersListManager.cs (hunk 2 above)
Comment (AngieDutra) [reply client/Assets/Scripts/UI/CharactersListManager.cs]: On it!

**Review by Nico-Sanchez — DISMISSED** [review]

File: client/Assets/Scripts/UI/CharactersListManager.cs [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        GenerateList();
+    }
+
+    void GenerateList()
+    {
+        players.ForEach(player =>
+        {
+            GameObject item = (GameObject)Instantiate(listItem, this.transform);
+            item.GetComponentInChildren<Image>().sprite = player.characterSprite;
+            item.GetComponentInChildren<TextMeshProUGUI>().text = player.name;
```
Comment (BertovDev) [inline client/Assets/Scripts/UI/CharactersListManager.cs:27]:
```suggestion
        CharactersScriptables.ForEach(character =>
        {
            GameObject item = Instantiate(listItem, this.transform);
            item.GetComponentInChildren<Image>().sprite = character.characterSprite;
            item.GetComponentInChildren<TextMeshProUGUI>().text = character.name;
```

**Review by BertovDev — CHANGES_REQUESTED** [review]

File: client/Assets/Prefabs/UI/CharacterListItem.prefab
[the remark this answers is not in the document]
Comment (AngieDutra) [reply client/Assets/Prefabs/UI/CharacterListItem.prefab:1]: Talked with our TL and agreed it wasn't wrong to do it in order to solve design issues. With that being said design changes were made and functionality changes resulted in this not being an issue anymore

File: client/Assets/Scripts/UI/CharactersListManager.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using TMPro;
+
+public class CharactersListManager : MonoBehaviour
+{
+    [SerializeField]
+    List<CoMCharacter> players;
```
[the remark this answers is not in the document]
Comment (AngieDutra) [reply client/Assets/Scripts/UI/CharactersListManager.cs]: I agree on this but changed it to CharacterScriptableObjects

File: client/Assets/Scripts/UI/CharactersListManager.cs (hunk 3 above)
Comment (AngieDutra) [reply client/Assets/Scripts/UI/CharactersListManager.cs]: This was already adressed

**Review by BertovDev — CHANGES_REQUESTED** [review]
Characters Button in mainScreen is not working.

**Comment by AngieDutra** [comment]
> Characters Button in mainScreen is not working.

Are you sure? I'm testing it and it works just fine to me. Make sure to update your branch and to close and open the Unity project

**Review by BertovDev — DISMISSED** [review]

**Review by tvillegas98 — APPROVED** [review]
LGTM!

**Comment by AngieDutra** [comment]
> I dont quite understand the `IsBackButton` functionality in `ButtonAnimationsMMTouchButton` can you please bring a little more explanation about it.

Sure! isBackButton is useful to identify back buttons. Although these should have the same looking scale animation due to its different size the scale difference should be bigger. It is also helpful to identify the buttons that trigger a change of scene to avoid the error of animating an element that is no longer there. I believe there is room for improvement tho, but this was the first implementation

**Comment by BertovDev** [comment]
> > I dont quite understand the `IsBackButton` functionality in `ButtonAnimationsMMTouchButton` can you please bring a little more explanation about it.
> 
> Sure! isBackButton is useful to identify back buttons. Although these should have the same looking scale animation, due to its different size the scale difference should be bigger to be noticeable. It is also helpful to identify the buttons that trigger a change of scene to avoid the error of animating an element that is no longer there. I believe there is room for improvement tho, but this was the first implementation

Nice!, thanks for the explanation

**Review by BertovDev — APPROVED** [review]
Nice!

**Pull request merged**